### DISCUSSION TITLE
Relation to serialization

### DISCUSSION THREAD

**Discussion by repli2dev** [CuyZ/Valinor#351] [Ideas]
Context: In various systems the API/Queue payload is hydrated into DTO, checks (& occasionaly changes) are done and then is it serialized again to the queue, logs... 

I understand that this library does not want to provide serialization as the complexity can rise significantly, yet some architectonic decisions make implementing symetric serialization more complicated.

Namely: key mapping & cammel support - it is splitted from the code and it is surjective (the camelCase names are still supported).

Could a better solution be considered such as configuring renaming on the particular field level that will be always used and accessed from the scope of serialization.

**Comment by romm** [comment]
Hello @repli2dev, sorry for the late answer. I've just submitted [a discussion in the repository](https://github.com/CuyZ/Valinor/discussions/420), I'd love to have your input on it, to better understand what people need. Hope you can join! 😊
